File: bucket_1/7.ts
import express from 'express';

enum EAccountState {
    STATE_UNSPECIFIED = 0,
    CLAIMED = 1,
    NOT_CLAIMED = 2
}

enum EAccountType {
    ACCOUNT_TYPE_UNSPECIFIED = 0,
    PARENT_MAIN = 1,
    PARENT_OTHER = 2,
    CHILD = 3
}

enum EListTweetsRequestListTweetsType {
    LIST_TWEET_TYPE_UNSPECIFIED = 0,
    USERS = 1,
    TRENDING = 2,
    MY_FOLLOWERS = 3,
    FOR_ME = 4
}

enum EListTweetsRequestSortType {
    SORT_TYPE_UNSPECIFIED = 0,
    RECENT = 1,
    RELEVANT = 2
}

interface Account {
    first_name?: string;
    last_name?: string;
    birth_day?: string;
    email?: string;
    username?: string;
    password?: string;
    account_type?: EAccountType;
    account_state?: EAccountState;
}

interface ClaimAccountRequest {
    account?: Account;
    parent_account_reference_token?: string;
}

interface CreateAccountRequest {
    account?: Account;
}

interface Tweet {
    id?: string;
    created_time?: Timestamp;
    message?: string;
    author?: string;
    like_count?: number;
    retweet_count?: number;
}

interface CreateTweetRequest {
    tweet?: Tweet;
}

interface ListTweetsResponse {
    tweets?: Tweet[];
    next_page_token?: string;
}

interface Timestamp {
    seconds: number;
    nanos: number;
}

const app = express();
app.use(express.json());

// Mock database
const accountsDb: Record<string, Account> = {};
const tweetsDb: Record<string, Tweet> = {};

// API endpoints
app.post('/v1/account', (req, res) => {
    const requestAccount: Account = req.body.account;
    const account: Account = {
        ...requestAccount
    };
    // TODO: Add account creation logic
    res.json(account);
});

app.patch('/v1/account/:username', (req, res) => {
    const username = req.params.username;
    const requestUpdate = req.body;
    const updateMask: string[] = requestUpdate.update_mask || [];
    // TODO: Update account logic
    res.json(accountsDb[username]);
});

app.post('/v1/account:verify', (req, res) => {
    const requestAccount = req.body.account;
    const accountVerficationToken = req.body.account_verfication_token;
    // TODO: Verify account logic
    res.json(requestAccount);
});

app.post('/v1/account:claim', (req, res) => {
    const claimRequest: ClaimAccountRequest = req.body;
    // TODO: Claim account logic
    res.json(claimRequest.account);
});

app.get('/v1/:parent/tweets', (req, res) => {
    const parent = req.params.parent;
    const listTweetsType = req.query.list_tweets_type as string;
    const sortType = req.query.sort_type as string;
    const pageSize = Number(req.query.page_size) || 50;
    const pageToken = req.query.page_token as string;
    // TODO: List tweets logic
    res.json({
        tweets: [],
        next_page_token: ''
    });
});

app.post('/v1/tweet', (req, res) => {
    const requestTweet: Tweet = req.body.tweet;
    const tweet: Tweet = {
        ...requestTweet
    };
    // TODO: Create tweet logic
    res.json(tweet);
});

app.delete('/v1/tweet/:tweet_id', (req, res) => {
    const tweetId = req.params.tweet_id;
    // TODO: Delete tweet logic
    res.sendStatus(204);
});

app.post('/v1/tweet/like', (req, res) => {
    const requestAccount = req.body.account;
    const requestTweet = req.body.tweet;
    // TODO: Like tweet logic
    res.sendStatus(204);
});

app.post('/v1/tweet/retweet', (req, res) => {
    const requestTweet: Tweet = req.body.tweet;
    // TODO: Retweet logic
    res.sendStatus(204);
});

app.listen(3000, () => {
    console.log('Server is running on port 3000');
});

File: bucket_1/7.py
from datetime import date
from enum import Enum
from typing import List
from flask import Flask, request, jsonify

app = Flask(__name__)

class EAccountState(Enum):
    STATE_UNSPECIFIED = 0
    CLAIMED = 1
    NOT_CLAIMED = 2

class EAccountType(Enum):
    ACCOUNT_TYPE_UNSPECIFIED = 0
    PARENT_MAIN = 1
    PARENT_OTHER = 2
    CHILD = 3

class EListTweetsRequestListTweetsType(Enum):
    LIST_TWEET_TYPE_UNSPECIFIED = 0
    USERS = 1
    TRENDING = 2
    MY_FOLLOWERS = 3
    FOR_ME = 4

class EListTweetsRequestSortType(Enum):
    SORT_TYPE_UNSPECIFIED = 0
    RECENT = 1
    RELEVANT = 2

class Account:
    def __init__(self, first_name=None, last_name=None, birth_day=None,
                 email=None, username=None, password=None, account_type=None, account_state=None):
        self.first_name = first_name
        self.last_name = last_name
        self.birth_day = birth_day
        self.email = email
        self.username = username
        self.password = password
        self.account_type = account_type
        self.account_state = account_state

class ClaimAccountRequest:
    def __init__(self, account=None, parent_account_reference_token=None):
        self.account = account
        self.parent_account_reference_token = parent_account_reference_token

class CreateAccountRequest:
    def __init__(self, account=None):
        self.account = account

class Tweet:
    def __init__(self, id=None, created_time=None, message=None, author=None,
                 like_count=None, retweet_count=None):
        self.id = id
        self.created_time = created_time
        self.message = message
        self.author = author
        self.like_count = like_count
        self.retweet_count = retweet_count

class CreateTweetRequest:
    def __init__(self, tweet=None):
        self.tweet = tweet

class ListTweetsResponse:
    def __init__(self, tweets=None, next_page_token=None):
        self.tweets = tweets
        self.next_page_token = next_page_token


accounts_db = {}
tweets_db = {}

# API endpoints
@app.route('/v1/account', methods=['POST'])
def create_account():
    request_data = request.get_json()
    account = Account(**request_data)
    # TODO: Add account creation logic
    return jsonify(account.__dict__)

@app.route('/v1/account/<username>', methods=['PATCH'])
def update_account(username):
    request_data = request.get_json()
    update_mask = request_data.get('update_mask', [])
    # TODO: Update account logic
    return jsonify(accounts_db[username].__dict__)

@app.route('/v1/account:verify', methods=['POST'])
def verify_account():
    request_data = request.get_json()
    account = request_data.get('account')
    account_verfication_token = request_data.get('account_verfication_token')
    # TODO: Verify account logic
    return jsonify(account.__dict__)

@app.route('/v1/account:claim', methods=['POST'])
def claim_account():
    request_data = request.get_json()
    claim_request = ClaimAccountRequest(**request_data)
    # TODO: Claim account logic
    return jsonify(claim_request.account.__dict__)

@app.route('/v1/<parent>/tweets', methods=['GET'])
def list_tweets(parent):
    list_tweets_type = request.args.get('list_tweets_type')
    sort_type = request.args.get('sort_type')
    page_size = request.args.get('page_size', default=50, type=int)
    page_token = request.args.get('page_token')
    # TODO: List tweets logic
    return jsonify(ListTweetsResponse(tweets=[]).__dict__)

@app.route('/v1/tweet', methods=['POST'])
def create_tweet():
    request_data = request.get_json()
    tweet = Tweet(**request_data)
    # TODO: Create tweet logic
    return jsonify(tweet.__dict__)

@app.route('/v1/tweet/<tweet_id>', methods=['DELETE'])
def delete_tweet(tweet_id):
    # TODO: Delete tweet logic
    return '', 204

@app.route('/v1/tweet/like', methods=['POST'])
def like_tweet():
    request_data = request.get_json()
    account = request_data.get('account')
    tweet = request_data.get('tweet')
    # TODO: Like tweet logic
    return '', 204

@app.route('/v1/tweet/retweet', methods=['POST'])
def retweet():
    request_data = request.get_json()
    tweet = request_data.get('tweet')
    # TODO: Retweet logic
    return '', 204

if __name__ == '__main__':
    app.run(debug=True)
